### DISCUSSION TITLE
TypeError: an integer is required

### DISCUSSION THREAD

**Discussion by raphaelschwinger** [PyAV-Org/PyAV#1666] [4. Bugs] [answered]
## Overview

I would like to output a video but get the follwoing error:

```python
import torch
import torchvision

# Example dummy video tensor
video = torch.randint(0, 255, (30, 720, 1280, 3), dtype=torch.uint8)  # 30 frames of 720p video

# Write the video
torchvision.io.write_video("test_video.mp4", video, fps=30, video_codec="h264")
```

Error:
```
File av/video/frame.pyx:193, in av.video.frame.VideoFrame.pict_type.__set__()

TypeError: an integer is required
```

## Expected behavior

Get a video file.

## Actual behavior

Error

## Versions

```
python = "^3.10"
torch = "^2.5.1"
torchvision = "^0.20.1"
lightning = "^2.4.0"
torchmetrics = "^1.5.1"
hydra-core = "^1.3.2"
hydra-colorlog = "^1.2.0"
hydra-optuna-sweeper = "^1.2.0"
rootutils = "^1.0.7"
pre-commit = "^4.0.1"
rich = "^13.9.4"
pytest = "^8.3.3"
torchrl = "^0.6.0"
tensordict = "^0.6.2"
ipykernel = "^6.29.5"
gymnasium = {extras = ["atari"], version = "<1.0.0"}
pygame = "^2.6.1"
av = "^14.0.0"

ffmpeg version 4.4.2-0ubuntu0.22.04.1
```

## Research

I have done the following:

- [x] Checked the [PyAV documentation](https://pyav.basswood-io.com)
- [x] Searched on [Google](https://www.google.com/search?q=pyav+how+do+I+foo)
- [x] Searched on [Stack Overflow](https://stackoverflow.com/search?q=pyav)
- [x] Looked through [old GitHub issues](https://github.com/PyAV-Org/PyAV/issues?&q=is%3Aissue)

**Comment by WyattBlue** [comment]
Go report this to torchvision

**Reply by raphaelschwinger** [reply]
https://github.com/pytorch/vision/issues/8779

**Comment by Zombiessss** [comment]
same error
<img width="917" alt="image" src="https://github.com/user-attachments/assets/6b226acd-29fd-42ca-a2f3-bf2107bf80f1">

**Comment by raphaelschwinger** [comment]
Version `13.1.0` works!

**Comment by WyattBlue** [comment, marked as answer]
Replace with.
```python
frame.pict_type = PictureType.NONE
# or
frame.pict_type = 0  
```

Major versions bumps are allowed to breaking.

**Reply by saltyfishxbr** [reply]
perfect! I met the same issue in torchrl VideoRecorder. Thanks!
